demisto_sdk/commands/common/tools.py

```diff
@@ -208,117 +208,143 @@ def get_core_pack_list() -> list:
+                               params={'ref': tag},
+                               headers={'PRIVATE-TOKEN': gitlab_token},
+                               verify=False)
             res.raise_for_status()
+        else:  # Github
+            res = requests.get(git_path, verify=False, timeout=10, headers={
+                'Authorization': f"Bearer {github_token}" if github_token else None,
+                'Accept': f'application/vnd.github.VERSION.raw',
+            })  # Sometime we need headers
+            if not res.ok:  # sometime we need param token
+                res = requests.get(
+                    git_path,
+                    verify=False,
+                    timeout=10,
+                    params={'token': github_token}
+                )
+
+        res.raise_for_status()
     except Exception as exc:
         # Replace token secret if needed
         err_msg: str = str(exc).replace(github_token, 'XXX') if github_token else str(exc)
         err_msg = err_msg.replace(gitlab_token, 'XXX') if gitlab_token else err_msg
         if not suppress_print:
+            if is_external_repository():
+                click.secho(
+                    f'You are working in a private repository: "{git_config.current_repository}".\n'
+                    f'The github/gitlab token in your environment is undefined.\n'
+                    f'Getting file from local repository instead. \n'
+                    f'If you wish to get the file from the remote repository, \n'
+                    f'Please define your github or gitlab token in your environment.\n'
+                    f'`export {git_config.credentials.ENV_GITHUB_TOKEN_NAME}=<TOKEN> or`\n'
+                    f'export {git_config.credentials.ENV_GITLAB_TOKEN_NAME}=<TOKEN>', fg='yellow'
+                )
+
             click.secho(
                 f'Could not find the old entity file under "{git_path}".\n'
                 'please make sure that you did not break backward compatibility.\n'
                 f'Reason: {err_msg}', fg='yellow'
             )
         return {}
-    file_content = res.content if res.ok else local_content
+    return get_file_details(res.text, full_file_path, return_content)
+
+
+def get_file_details(
+        file_content: str,
+        full_file_path: str,
+        return_content: bool = False,
+):
     if return_content:
-        return file_content
-    if full_file_path.endswith('json'):
-        details = res.json() if res.ok else json.loads(local_content)
+        file_details = file_content
+    elif full_file_path.endswith('json'):
+        file_details = json.loads(file_content)
     elif full_file_path.endswith('yml'):
-        details = ryaml.load(file_content)
+        file_details = ryaml.load(file_content)
     # if neither yml nor json then probably a CHANGELOG or README file.
     else:
-        details = {}
-    return details
+        file_details = {}
+    return file_details
+
+
+# @lru_cache(maxsize=64)
```

Comment:
can we enable cache?

Reply:
I don't see why we can't do that. That's line was commented before, will investigate why.